### ISSUE TITLE
High CPU load with any browser

### ISSUE THREAD

**Issue by cmonty14** [bigbluebutton/bigbluebutton#11582]
**Describe the bug**
My system is running an updated Arch Linux installation.
When I join a BBB meeting my system shows high CPU load independent of the browser (Firefox, Chromium, Vivaldi).
If I turn off my webcam, the CPU load remains high.

**To Reproduce**
1. Join BBB meeting with video
2. Monitor CPU load

**Expected behavior**
Less CPU load.

**Actual behavior**
CPU load of all cores is 95-100%.

**Screenshots**
![screenshotCPU](https://user-images.githubusercontent.com/12523591/110328982-4fba9680-801c-11eb-9642-4a535a4e7125.png)

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Browser: Firefox, Chromium, Vivaldi
 - Version: latest

**Comment by antobinary** [comment]
Hi @cmonty14 ,

Thank you for reporting this! Could you please try to grab a screenshot of the same scenario on Chrome/Chromium with the Task Manager open? (Shift + Esc to activate it in Chromium) That will show the CPU / memory usage for the BBB tab.

![Screenshot from 2021-03-08 09-42-37](https://user-images.githubusercontent.com/6312397/110336403-d490ba80-7ff2-11eb-9a4f-ac89f7102763.png)

Also, do you join a session with only one webcam (yours) or someone else's session?

**Comment by antobinary** [comment]
Other useful info would be:
* what version of BigBlueButton do you see this on - 2.2.35 or perhaps 2.3-alpha7?
* if you can try to reproduce this on https://test23.bigbluebutton.org/demo/demo2.jsp

**Comment by basisbit** [comment]
Also, within Chrome/Chromium please provide a screenshot from the `chrome://gpu/` overview

**Comment by cmonty14** [comment]
Hi,
I have created a screenshot of the task manager.
![signal-2021-03-09-080316_002](https://user-images.githubusercontent.com/12523591/110432290-3151ab00-80af-11eb-9c75-6f4f4b71e9de.png)
When this screenshot was created, BBB was used in a classroom session with 12 participants. Each participant is using his/her webcam + 1 participant sharing his/her desktop.
Typically 20 participants are joining the same session.

Can you advise how to identify the BBB version?
If this version is only available on the server, it would be difficult because I don't have access to the server(s), I'm only using the (web browser) client. Here I can find this information: _Clientversion: 1114_

In addition I have created another screenshot with the CPU utilization (htop) and information of CPU and graphic card of my desktop PC.
![screenshotTerminal](https://user-images.githubusercontent.com/12523591/110433106-5561bc00-80b0-11eb-8c62-27537b27482d.png)

![signal-2021-03-09-080316_001](https://user-images.githubusercontent.com/12523591/110433171-66aac880-80b0-11eb-8380-034fd8eb5aac.png)

Furthermore I pasted the output of the `chrome://gpu/` overview into an editor and uploaded the file here.
[gpu.txt](https://github.com/bigbluebutton/bigbluebutton/files/6106618/gpu.txt)

THX for supporting this issue.

**Comment by cmonty14** [comment]
I can reproduce the high CPU load on https://test23.bigbluebutton.org/demo/demo2.jsp selecting courses "English 232" and "English 402".
![2021-03-09_09-05](https://user-images.githubusercontent.com/12523591/110439168-6b26af80-80b7-11eb-88f3-9187c1c89abe.png)
![2021-03-09_09-07](https://user-images.githubusercontent.com/12523591/110439184-7083fa00-80b7-11eb-8414-65d2408346ad.png)

**Comment by basisbit** [comment]
1114 is bbb version 2.2.31. I'd guess that video-pagination and cameraQualityThresholds are not enabled on that server. Thus, an update to a current version of BigBlueButton should probably resolve the problem.

**Comment by prlanzarin** [comment]
> 1114 is bbb version 2.2.31. I'd guess that video-pagination and cameraQualityThresholds are not enabled on that server. Thus, an update to a current version of BigBlueButton should probably resolve the problem.

@cmonty14 can reproduce the issue in test23 so I doubt pagination or dynamic profiles will make any difference on that regard.

@cmonty14 care to share your CPU specs?

**Comment by prlanzarin** [comment]
@cmonty14 Also, do a comparison by opening two tabs of https://appr.tc/ (with the same room name).
Attach info screenshots as you did with BBB, please.

**Comment by cmonty14** [comment]
> @cmonty14 care to share your CPU specs?
What exactly do you request?
Please advise how to generate the CPU specs.

**Comment by cmonty14** [comment]
> > 1114 is bbb version 2.2.31. I'd guess that video-pagination and cameraQualityThresholds are not enabled on that server. Thus, an update to a current version of BigBlueButton should probably resolve the problem.
> 
> @cmonty14 can reproduce the issue in test23 so I doubt pagination or dynamic profiles will make any difference on that regard.
> 
> @cmonty14 care to share your CPU specs?

What exactly do you request?
Please advise how to generate the CPU specs.

**Comment by cmonty14** [comment]
This is the hardware I'm using.
https://www.msi.com/Desktop/Cubi-3-Silent-7m/Specification

**Comment by cmonty14** [comment]
> @cmonty14 Also, do a comparison by opening two tabs of https://appr.tc/ (with the same room name).
> Attach info screenshots as you did with BBB, please.

The CPU load is very high >200 in Chromium Task Manager.
![signal-2021-03-10-084539_001](https://user-images.githubusercontent.com/12523591/110594536-3715c200-817d-11eb-9280-86bd39dd42f1.png)
![signal-2021-03-10-084539_002](https://user-images.githubusercontent.com/12523591/110594566-409f2a00-817d-11eb-872c-1869a86c1eab.png)

**Comment by prlanzarin** [comment]
@cmonty14 

tl;dr: this seems out of our reach. The **short term mitigation** I recommend to you is enabling `**Data savings**` (three dots [client's top right] -> Settings -> Data savings -> Toggle "Enable webcams" and "Enable desktop sharing" off")

It looks like to me as if this is just a case of the underlying libwebrtc usage in Chromium or Gecko-based browsers requiring a bit more CPU power than what your machine's capable of giving. How efficient media is in BBB (and browser based WebRTC apps in general) usually depend on various factors, but the main ones usually are: architecture and CPU model it's running in, dedicated HW encoding/decoding availability, used codecs, browser's build environment, browser's running envs and its dependencies, ... The fact that the same behaviour happens in appr.tc (a P2P sample, if not canonical, WebRTC app) kind of clear things up.
It might be that something in Arch Linux builds of gecko/chromium is borking media transcoding efficiency, but I doubt it. You'd have to fiddle with that.

I'd say the best theoretical bet to lower that CPU usage would be to set H.264 as the default codec and force its profile ID in BBB to be the one browser's usually do best with HW de/enconding. But our media server currently sucks with H.264 so we can't really recommend that until we swap it for a media server that actually works with H.264.

**Comment by basisbit** [comment]
enabling h.264 however makes BBB unusable on older hardware without hardware acceleration for h264, however the current setup with vp8 works fine with quite old hardware, even with 20 enabled webcams in Chromium on Windows, as long as `cameraQualityThresholds ` is enabled.

**Comment by prlanzarin** [comment]
>enabling h.264 however makes BBB unusable on older hardware without hardware acceleration for h264, however the current setup with vp8 works fine with quite old hardware, even with 20 enabled webcams in Chromium on Windows, as long as cameraQualityThresholds is enabled.

For sure it works fine, and better than H.264. That's why we've chosen VP8 as the mainline default. I'm **not** suggesting H.264 as default of mainline BBB if that wasn't clear; I was talking about testing H.264 as default in a specific test installation (but, as I said, I can't even recommend doing that test due to server-side issues).

Again, my only real suggestion right now is to enable data savings.

**Comment by cmonty14** [comment]
> @cmonty14
> 
> tl;dr: this seems out of our reach. The **short term mitigation** I recommend to you is enabling `**Data savings**` (three dots [client's top right] -> Settings -> Data savings -> Toggle "Enable webcams" and "Enable desktop sharing" off")
> 
> It looks like to me as if this is just a case of the underlying libwebrtc usage in Chromium or Gecko-based browsers requiring a bit more CPU power than what your machine's capable of giving. How efficient media is in BBB (and browser based WebRTC apps in general) usually depends on various factors, but the main ones usually are: architecture and CPU model it's running in, dedicated HW encoding/decoding availability, used codecs, browser's build environment, browser's running envs and its dependencies, ... The fact that the same behaviour happens in appr.tc (a P2P sample, if not canonical, WebRTC app) kind of clear things up.
> It might be that something in Arch Linux builds of gecko/chromium is borking media transcoding efficiency, but I doubt it. You'd have to fiddle with that.
> 
> I'd say the best theoretical bet to lower that CPU usage would be to set H.264 as the default codec and force its profile ID in BBB to be the one browser's usually do best with HW de/enconding. But our media server currently sucks with H.264 so we can't really recommend that until we swap it for a media server that actually works with H.264.

Can you please advise how to set/enable data savings? 
I'm checking with Chrome, but cannot find this function in settings.
Maybe a screenshot could help.

**Comment by antobinary** [comment]
These are settings within the BigBlueButton client

![Screenshot from 2021-03-12 05-27-00](https://user-images.githubusercontent.com/6312397/110927743-03a97380-82f4-11eb-9370-e9261ff95e09.png)
![Screenshot from 2021-03-12 05-27-10](https://user-images.githubusercontent.com/6312397/110927758-0906be00-82f4-11eb-8720-f9cd7b07d9de.png)

The idea was to disable (at least) the webcams

**Comment by basisbit** [comment]
I don't see anything left to do or how we could change BBB to improve this when the webbrowser/drivers/hardware don't deliver the required performance. Should we thus close this issue?
